CONTRIBUTING.md

```diff
@@ -27,5 +31,48 @@ I recommend using [Visual Studio Code](https://code.visualstudio.com/) with
 
 
 
+## High-Level Documentation
+
+### Important Terms
+
+- CAS: Computer Algebra System, this is a piece of magic that takes formulas or equations and evaluates them
+- MathJson: [A format](https://cortexjs.io/math-json/) to unambiguously represent mathematical expressions
+- Document: A document consists of a number of elements such as
+  - Mathematical Expressions
+  - Scopes
+  - Text Blocks
+
```

Comment:
Is the current design anywhere near fine? Parts of it seem pretty darn complicated, like the type definitions for the document https://github.com/stefnotch/quantum-sheet/blob/master/src/model/document/document.ts

On that note, another complicated thing is how scopes are implemented